chocolatey/plugins/module_utils/Packages.psm1

```diff
@@ -960,120 +960,142 @@ function Install-Chocolatey {
     $chocoCommand = Get-ChocolateyCommand -IgnoreMissing
     if ($null -eq $chocoCommand) {
         # We need to install chocolatey
-        # Enable necessary TLS versions if they're available but disabled.
-        # Default for win_chocolatey is to allow TLS 1.1, 1.2, and 1.3 (if available)
-        $protocols = [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::SystemDefault
-
-        foreach ($tlsVersion in $BootstrapTlsVersion) {
-            # If the TLS version isn't available on the system, this will evaluate to $null and be skipped
-            $value = $tlsVersion -as [System.Net.SecurityProtocolType]
-            if ($value) {
-                $protocols = $protocols -bor $value
-            }
+
+        # Chocolatey CLI v2.0.0 and above requires .NET Framework 4.8 to be installed.
+        # If the user has specified a 1.x version of Chocolatey to install, or the .NET requirement is met, we'll install Chocolatey.
+        $dotNetRegistryPath = "HKLM:\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full"
+        $installedDotNetVersion = [version]((Get-ItemProperty -Path $dotNetRegistryPath -Name Version).Version)
+
+        $chocolateyLegacyVersion = if ($Version) {
+            [version]$Version -lt [version]"2.0.0"
+        } else {
+            $false
         }
 
-        [System.Net.ServicePointManager]::SecurityProtocol = $protocols
+        if (($chocolateyLegacyVersion) -or ($installedDotNetVersion -ge [version]"4.8")) {
```

Comment:
We should invert this if, so that we reduce nesting for the rest of this function:

```ps1
if (-not $chocolateyLegacyVersion -and $installedDotNetVersion -lt [version]'4.8') {
    # throw error
}

# continue with the remainder of the function
```

`Assert-TaskFailed` will terminate execution much like `exit` as it calls the `FailJson()` method on the ansible module, so there's no need to worry about early return or whatnot.

Reply:
Good call, at the time I didn't know what was causing execution to terminate so "played it safe", will switch that up.